snuba/datasets/configuration/json_schema.py

```diff
@@ -213,3 +214,17 @@ def make_column_schema(
         "required_time_column",
     ],
 }
+
+V1_DATASET_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "version": {"const": "v1"},
+        "kind": {"const": "dataset"},
```

Comment:
Shouldn't this go in `KIND_SCHEMA`?

Reply:
Removed `KIND_SCHEMA` entirely since we don't want a dataset to be able to have `entity` as a kind etc. Each kind should be a const as it is now.